tests/ballerina-test/src/test/java/org/ballerinalang/test/jwt/JwtTest.java

```diff
@@ -0,0 +1,121 @@
+    private static final String KEY_STORE_TYPE = "type";
+    private static final String KEY_STORE_PASSWORD = "keyStorePassword";
+    private static final String TRUST_STORE_CONFIG = "trustStore";
+    private static final String TRUST_STORE_LOCATION = "location";
+    private static final String TRUST_STORE_TYPE = "type";
+    private static final String TRUST_STORE_PASSWORD = "trustStorePassword";
+
+    private ConfigRegistry initialConfigRegistry;
+    private String resourceRoot;
+    private CompileResult compileResult;
+    private String jwtToken;
+
+    @BeforeClass
+    public void setup() throws Exception {
+        initialConfigRegistry = ConfigRegistry.getInstance();
+
+        ConfigRegistry configRegistry = mock(ConfigRegistry.class);
+        when(configRegistry.getInstanceConfigValue(KEY_STORE_CONFIG, KEY_STORE_LOCATION))
+                .thenReturn(getClass().getClassLoader().getResource(
+                        "datafiles/security/keyStore/ballerinaKeystore.p12").getPath());
+        when(configRegistry.getInstanceConfigValue(KEY_STORE_CONFIG, KEY_STORE_PASSWORD))
+                .thenReturn("ballerina");
+        when(configRegistry.getInstanceConfigValue(KEY_STORE_CONFIG, KEY_STORE_TYPE))
+                .thenReturn("pkcs12");
+        when(configRegistry.getInstanceConfigValue(TRUST_STORE_CONFIG, TRUST_STORE_LOCATION))
+                .thenReturn(getClass().getClassLoader().getResource(
+                        "datafiles/security/keyStore/ballerinaTruststore.p12").getPath());
+        when(configRegistry.getInstanceConfigValue(TRUST_STORE_CONFIG, TRUST_STORE_PASSWORD))
+                .thenReturn("ballerina");
+        when(configRegistry.getInstanceConfigValue(TRUST_STORE_CONFIG, TRUST_STORE_TYPE))
+                .thenReturn("pkcs12");
+
+        Field field = ConfigRegistry.class.getDeclaredField("configRegistry");
+        field.setAccessible(true);
+        Field modifiersField = Field.class.getDeclaredField("modifiers");
+        modifiersField.setAccessible(true);
+        modifiersField.setInt(field, field.getModifiers() & ~Modifier.FINAL);
+        field.set(ConfigRegistry.class, configRegistry);
+        modifiersField.setInt(field, field.getModifiers() & Modifier.FINAL);
+        modifiersField.setAccessible(false);
+        field.setAccessible(false);
+
+        resourceRoot = getClass().getProtectionDomain().getCodeSource().getLocation().getPath();
+        Path sourceRoot = Paths.get(resourceRoot, "test-src", "jwt");
+        compileResult = BCompileUtil.compile(sourceRoot.resolve("jwt-test.bal").toString());
+    }
+
+    @Test(priority = 1, description = "Test case for issuing JWT token with valid data")
+    public void testIssueJwt() throws Exception {
+        BValue[] returns = BRunUtil.invoke(compileResult, "testIssueJwt");
+        Assert.assertTrue(returns[0] instanceof BString);
+        Assert.assertNull(returns[1]);
+        jwtToken = returns[0].stringValue();
```

Comment:
Here we set value for jwtToken but it doesn't have any further usages. Am I missing something here?

Reply:
this is the JWT token that used to test verify method.